orion/sites/admin/src/home/home.js

```diff
@@ -2,30 +2,210 @@
 require("./home.styl");
 import React from "react";
 import ReactDOM from "react-dom";
+import API from "../api.js";
 
 export class HomePage extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            programs: [],
+            classes: [],
+            locations: [],
+            achieves: [],
+            semesters: [],
+            sessions: [],
+            row: [],
+            selection: "",
+            num: 0,
+            noUnpub: "",
+            switch: false,
+        };
+        this.onClickPrograms = this.onClickPrograms.bind(this);
+        this.onClickClasses = this.onClickClasses.bind(this);
+        this.onClickLocations = this.onClickLocations.bind(this);
+        this.onClickAchievements = this.onClickAchievements.bind(this);
+        this.onClickSemesters = this.onClickSemesters.bind(this);
+        this.onClickSessions = this.onClickSessions.bind(this);
+    }
+
+    componentDidMount() {
+        API.get("api/v1/unpublished").then((res) => {
+            const unpublishedList = res.data;
+            this.setState({
+                programs: unpublishedList.programs,
+                classes: unpublishedList.classes,
+                locations: unpublishedList.locations,
+                achieves: unpublishedList.achieves,
+                semesters: unpublishedList.semesters,
+                sessions: unpublishedList.sessions,
+            });
+        });
+    }
+
+    onClickPrograms() {
+        this.setState({ noUnpub: "" });
+        let programs = this.state.programs.map((row, index) => {
+            return <DashboardRow key={index} row={row} />;
+        });
+        this.setState({ row: programs });
+        this.setState({ selection: "programs" });
+        this.setState({ num: programs.length });
```

Comment:
it looks like num is always equal to `this.state.row.length` right? 
In this case, just use that instead of having to keep track of another state variable (num). You can easily get into a state where you forget to update num and then your states become messed up.